### PR TITLE
add hasTable to Database

### PR DESCRIPTION
**Pull request by retorquere**

### CODE REVIEW

**Comment by dmfay** [comment]
Few things!

* I'm not sure why you changed the name -- there's nothing specific to document tables, so the original `hasTable` is more appropriate
* The documentation comment mentions creating the table instead of searching
* There's nothing async so it doesn't need to be a co-wrapped generator (the build is failing because it's a generator without a `yield`)
* Please add a testcase in test/database.

**Comment by retorquere** [comment]
1. I changed the name because I figured it's an companion method to `createDocumentTable`
2. Oops.
3. Oops again.
4. I have made an attempt but I'm not familiar with this test setup.

File: lib/database.js [hunk 1]
```diff
@@ -232,6 +232,22 @@ Massive.prototype.createDocumentTable = co.wrap(function* (path) {
...
+ * @param {String} path - Name of the table to find. Unless the schema is
+ * specified in a qualified path, 'public' is assumed.
+ * @return {Boolean} whether the table is known to the database
+ */
+Massive.prototype.hasDocumentTable = function (path) {
+  const splits = path.split(".");
+  const tableName = splits.pop();
+  const schemaName = splits.pop() || "public";
+
+  return this.tables.some((table) => (table.schema == schemaName && table.name == tableName))
```
Comment (dmfay) [inline lib/database.js]: eslint is complaining because this line is missing a semicolon.

Comment (retorquere) [inline lib/database.js]: Beat you to it :)

**Comment by retorquere** [comment]
I'm very sure the test I amended is naive BTW.

**Comment by dmfay** [comment]
1. It's just as much a companion to `dropTable` as it is to `createDocumentTable` imo.
4. Are you familiar with mocha? I've offloaded the bulk of the setup/teardown stuff into test/helpers/index.js so each test file is pretty lightweight. What I'd suggest is copying test/database/dropTable.js to test/database/hasTable.js and editing it to test your function instead. You want to assert that it comes back with `true` when the name is found and `false` when it isn't, both with a schema and without.

**Comment by retorquere** [comment]
I'm not familiar with mocha. What should I do to run the tests locally?

**Comment by dmfay** [comment]
You'll need PostgreSQL installed, and to create a database named `massive`. The built-in `postgres` user should be able to connect through peer authentication rather than a password (or you can temporarily change the connection string it uses in test/helpers/index.js, just be sure not to check it in!). Then `npm install` to download the dependencies, and `npm test` to run all the tests and linter. You can run just your test by `mocha path/to/file`.

Mocha is fairly simple, especially since most of the boilerplate is taken care of; `describe` methods group the `it('should...')` test methods, which in turn should throw AssertionErrors on failing. Async tests should return a promise or have a `it('should...', function (done) {...})` signature and invoke `done` on completion. Massive's tests include the [chai assertion library](http://chaijs.com/api/assert/) so there are a lot of ways to express what should happen. Your tests could take advantage of `assert.isTrue` and `assert.isFalse` for example.

**Comment by retorquere** [comment]
OK, separate tests, running green.

**Comment by dmfay** [comment]
Looks good, thanks!

**Comment by retorquere** [comment]
Should I have used this

```
  if (splits.length > 1) {
    schemaName = splits[0];
    tableName = splits[1];
    potentialTable = this[schemaName][tableName];
  } else {
    potentialTable = this[tableName];
  }
```

rather than the scan through `tables`?

**Comment by dmfay** [comment]
`this[name]` could just as easily be a view or function. `saveDoc` uses it because it has to invoke a function, but you're just looking to make sure it 1) exists and 2) is a table. You could have used `instanceof` to check its type, but it's kind of six of one, half a dozen of the other imo.

**Comment by retorquere** [comment]
But then perhaps `hasTable` should be `getTable` and use `find` rather than `some`. Then `saveDoc` could also use it to call `potentialTable.saveDoc(doc)`.

**Comment by dmfay** [comment]
If you want to refactor that, sure!

**Comment by retorquere** [comment]
No problem. Do you have any clues on how I can get my checkout synced to your github repo? I need to incorporate the merge commit. This is one of the things with git which always eludes me.

**Comment by dmfay** [comment]
You need to add this repo as another remote and pull from it:

```
git remote add dmfay git@github.com:dmfay/massive-js
git pull dmfay promises
```

**Pull request merged**